src/plugins/discover/public/application/view_components/canvas/index.tsx

```diff
@@ -171,6 +172,7 @@ export default function DiscoverCanvas({ setHeaderActionMenu, history, optionalR
             (isEnhancementsEnabled ? (
               <>
                 <MemoizedDiscoverChartContainer {...fetchState} />
+                <MemoizedDiscoverVisualization {...fetchState} />
```

Comment:
@d-buckner i can expect that the fetchState contain all the information for a aggregated PromQL returned data right? especially `rows: OpenSearchHit[]` of the `fetchState` here is the data i will use to feed into the visualization render logics

<img width="1722" alt="Screenshot 2025-03-06 at 2 11 32 PM" src="https://github.com/user-attachments/assets/da0e6a15-9a87-4d40-bda8-bcd36972f7d3" />

Reply:
also i can expect to be able to find the Prometheus data source information(id, schemas of the fields) through the data frame services, correct? 

```
/**
 * A data frame is a two-dimensional labeled data structure with columns of potentially different types.
 */
export interface IDataFrame {
  type?: DATA_FRAME_TYPES.DEFAULT;
  name?: string;
  schema?: Array<Partial<IFieldType>>;
  meta?: Record<string, any>;
  fields: IFieldType[];
  size: number;
}
```

Reply:
that schema isn't really used anymore. depends on the prom said if they will returning back the schema with the row/col where the schema is the information about the source regardless of the query or resource.

i would propose still relying on the index pattern fields. the temporary index pattern should fetch the fields in the dataset config. 

that index pattern could point to a metric it's abstracted out enough

Reply:
make sense. @d-buckner can we make sure that the metrics are created and mapped using the temporary index pattern flow like PPL?